modules/standard/Memory/Diagnostics.chpl

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright 2020-2021 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ * 
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * 
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  The :mod:`Diagnostics` module provides procedures which report information
+  about memory usage.  With one exception, to use these procedures you
+  must enable memory tracking.  Do this by setting one or more of the
+  config vars below, using appropriate ``--configVarName=value`` or
+  ``-sconfigVarName=value`` command line options when you run the
```

Comment:
Might be nice to link to the spec section on config params and config vars here, but I'd be okay with having that be a future work item